src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/RazorProjectHostBase.cs

```diff
@@ -249,6 +279,130 @@ Task IProjectDynamicLoadComponent.UnloadAsync()
             return DisposeAsync();
         }
 
+        private void ProjectManager_Changed(object sender, ProjectChangeEventArgs args)
+        {
+            switch (args.Kind)
```

Comment:
I would have imagine that [requirement 3](https://github.com/dotnet/aspnetcore/issues/17799#issuecomment-599859285) would exist somewhere here?

Reply:
talked offline, turns out requirement 3 is already satisfied.